data_pat.go

```diff
@@ -41,3 +41,27 @@ func parsePATSection(i *astikit.BytesIterator, offsetSectionsEnd int, tableIDExt
 	}
 	return
 }
+
+func (p *PATData) Serialise(b []byte) (int, error) {
+	currentIdx := 0
+	for i := range p.Programs {
+		n, err := p.Programs[i].Serialise(b[currentIdx:])
```

Comment:
With this paradigm, do bad things happen if we run off the end of `b`? Should there be a check for that?

Reply:
Yes bad things would happen. Have amended all the leaf functions to return errors which would bubble up.